cli/src/deposit.tsx

```diff
@@ -239,6 +241,106 @@ export const DepositControl = () => {
     ]
   )
 
+  const cli = useCli()
+
+  const pool = useOption('pool')
+
+  const asset = useOption('asset')
+
+  const amountArg = useOption('amount')
+
+  const { pairs } = usePools()
+
+  useEffect(() => {
+    if (!cli || !pool || !pairs || stage !== Stage.PoolSelection) {
+      return
+    }
+
+    const pair = pairs.find((p) =>
+      [
+        `${p.token0}/${p.token1}`.toLowerCase(),
+        `${p.token1}/${p.token0}`.toLowerCase()
+      ].includes(pool.toLowerCase())
+    )
+
+    if (pair === undefined) {
+      addMessage(`Pool not found: ${pool.toUpperCase()}`, { color: 'red' })
+      setStage(Stage.Empty)
+      return
+    }
+
+    setChosenPair(pair)
+    setStage(Stage.CollateralSelection)
+  }, [cli, addMessage, pairs, pool, stage])
+
+  useEffect(() => {
+    if (
+      !cli ||
+      !asset ||
+      !chosenPair ||
+      !chosenPairInfo.ready ||
+      stage !== Stage.CollateralSelection
+    ) {
+      return
+    }
+
+    let choice = 0
+
+    if (chosenPair.token0.toLowerCase() === asset.toLowerCase()) {
+      choice = 1
+    } else if (chosenPair.token1.toLowerCase() === asset.toLowerCase()) {
+      choice = 2
+    }
+
+    if (choice === 0) {
+      addMessage(`Invalid collateral: ${asset.toUpperCase()}`, { color: 'red' })
+      setStage(Stage.Empty)
+      return
+    }
+
+    onCollateralSelected(choice)
+  }, [
+    cli,
+    stage,
+    chosenPair,
+    asset,
+    onCollateralSelected,
+    addMessage,
+    chosenPairInfo.ready
+  ])
+
+  useEffect(() => {
+    if (!cli || amountArg === undefined || stage !== Stage.AmountInput) {
+      return
+    }
+
+    if (isNaN(Number(amountArg))) {
+      addMessage(`Invalid amount: ${amountArg}`, { color: 'red' })
+      setStage(Stage.Empty)
+    }
+
+    onAmountSubmitted(String(amountArg))
+  }, [onAmountSubmitted, amountArg, cli, stage, addMessage])
+
+  useEffect(() => {
+    if (stage === Stage.Confirm) {
```

Comment:
meant for cli only?

Reply:
Right.
Sorry, I missed it and will fix now